### ISSUE TITLE
Cowin Public API

### ISSUE THREAD

**Issue by CodeWithBishal** [cowinapi/developer.cowin#289]
Hello, I want to fetch vaccination details using public API "calendarByPin" and build a website like (https://www.cowin.gov.in/home) where people can put their pin code and all the details will be available for their pin code. But the problem is "limit of 100 API calls per 5 minutes per IP" i.e. my website can call only 100 times per 5 minutes which might be a problem and the limit might get crossed within a minute as people will search with their pin code. I have the code ready, is it possible to host it on the digital ocean Bangalore server? and how can I build the website without API call restrictions??

**Comment by kichappa** [comment]
I don't think it is possible unless you have an approved backdoor (API key of some sort).

**Comment by CodeWithBishal** [comment]
how can I get that? and what is the use of Co-WIN protected API and how to get access

**Comment by kichappa** [comment]
API Setu declares,
> Co-WIN APIs to register for vaccination, to book a vaccination appointment from partner applications. These APIs have restricted access. [...]. **_Please contact the Ministry of Health and Family Welfare, Government of India to get access to these APIs._** 

You can use the protected API in many ways. I have described one of the ways in #278.

**Comment by CodeWithBishal** [comment]
@kichappa 
(1)what qualifies someone as a company/organization?
(2) Does it work on the digital ocean Bangalore server?

The guide you sent me is for fetching beneficiary list

**Comment by kichappa** [comment]
> 1. What qualifies someone as a company/organization?
> 2. Does it work on the digital ocean Bangalore server?

I am not sure. I am not affiliated with CoWin in any way. You should try enquiring it with the respective authorities.

> The guide you sent me is for fetching the beneficiary list.

The method can be used to `get` real-time data by changing a line or two. The authentication process can be the same.

**Comment by CodeWithBishal** [comment]
@kichappa Thanks for your help I have mailed them. Let's wait for a reply from them

**Comment by CodeWithBishal** [comment]
@kichappa If I use the public API for my project do I need to take permission?

**Comment by kichappa** [comment]
> Do I need to take permission?

No. Even protected APIs are free to use by Indian citizens. There'll just be API limitations (no more than 100 queries in 5 minutes).

**Comment by CodeWithBishal** [comment]
Okay got it....thanks
